ion_hasher.go

```diff
@@ -22,6 +22,8 @@ type IonHasher interface {
 	// It never returns an error.
```

Comment:
That statement is bold. Are we not returning errors but we can fail with errors? the docs for public interfaces and methods will need some work.

Reply:
My apologies, I copied the docstrings from https://golang.org/src/hash/hash.go?s=1238:1887#L16

You are correct, we are returning errors because `Update(bytes []byte)` will call `ch.hashAlgorithm.Write(p)` which returns a tuple (n int, err error)

Reply:
@justing-bq Please remove:

> // It never returns an error.

Reply:
Removed.